crates/backend/src/codegen.rs

```diff
@@ -847,6 +855,15 @@ impl ToTokens for ast::ImportEnum {
                 }
             }
 
+            #[allow(clippy::all)]
+            impl ::core::str::FromStr for #name {
+                type Err = ();
```

Comment:
I'd personally prefer that if we auto-generate this impl that a more appropriate `Err` type is selected for use here.

Reply:
Sure; I can make this change.

What would an appropriate `Err` type be? Something like:

```rust
struct StringEnumUnknownVariant<'s> {
  pub valid_variants: &'static [&'static str],
  pub received: &'s str,
}
```

or just a `str` with the invalid variant string or something else?

Reply:
I haven't really thought about what the best error type would be, but designing good error types is unfortunately not trivial. I would prefer if we didn't auto-generate this for now and left it to users to implement if they need it.

Reply:
:+1: I'll remove it from this PR